docs/additional/internal_aspects_de.md

```diff
@@ -0,0 +1,289 @@
+}
+```
+
+[trivy]: https://github.com/cloudogu/dogu-build-lib/blob/develop/docs/development/trivy_de.md
+
+#### Manuelle Tests auf einer Test-Stage
+
+In der manuellen Testphase sollte das Dogu auf einer Test-Stage getestet werden.
+Dabei sollte die Test-Stage eine möglichst reale Umgebung abbilden.
+Das Dogu soll mit dem Befehl `cesapp install` installiert werden.
+Das Dogu sollte systematisch im Browser manuell getestet und als funktionstauglich eingestuft werden.
+Dabei sollte der Fokus auf den Features liegen, welche in Produktion die größte Relevanz haben.
+
+## Definition of Done
+
+Dieser Abschnitt beschreibt mögliche Voraussetzungen für einen finalen Zustand einer Entwicklung.
+Er schafft ein gemeinsames Verständnis für Entwickler wann ein Feature fertig ist und kann verschiedenste Anforderungen
+beinhalten.
+
+Folgende Punkte beschreiben grundlegende Anforderungen, die als Basis einer Definition of Done dienen können.
+
+### Issue-Tracking
+
+- Für jede Änderung existiert ein Issue in einem Ticketsystem.
+- Es beschreibt das zu entwickelnde Produktinkrement.
+- Tickets in öffentlichen Ticketsystem werden in internen Systemen verlinkt.
+
+### Dockerfile
+
+Anforderung für das Dockerfile sind [hier](interesting_aspects_de.md#container-images) zu finden.
+
+### Dogu.json
+
+- Muss konfigurierbare Schlüssel enthalten:
+  -  `container_config/memory_limit`
+  -  `container_config/swap_limit`
+
+### Dogu-Scripting
+
+Anforderungen für Shell-Scripting sind [hier](interesting_aspects_de.md#bash-skripte) zu finden.
+
+### Dogu-Funktionalitäten
+
+- Backup & Restore-Fähigkeit betrachten
+- Upgrade-Fähigkeit betrachten
+  - Upgrade-Skripte müssen so gestaltet sein, dass aus allen alten Versionen des Dogus ein Upgrade möglich ist 
+  (sofern das die im Dogu eingesetzte Software nicht verhindert)
+  - Ein Upgrade des Dogus von früherer Version ist zu testen
+- Admin-Gruppe
+  - Admin-Gruppe muss änderbar sein
+  - Admin-Gruppe aus dem `etcd` wird verwendet und bei einem Wechsel der Admin-Gruppe reagiert das Dogu entsprechend
+  darauf, dann wird der alten Admin-Gruppe das Admin-Recht entzogen. Siehe z.B. SonarQube und Nexus Repository Manager.
+- Logging
+  - Log-Output des Dogus auf Fehler checken
+- Übernahme einer Änderung der CES-FQDN
+  - Eine FQDN-Änderung muss in einem Dogu sinnvoll verarbeitet werden.
+  - Das Dogu muss nach einem Restart ordnungsgemäß starten.
+- Single Sign-on & Single Sign-out betrachten
+
+### Dokumentation
+
+- Dogu-Dokumentation aktualisieren und prüfen, ob diese dem [Dokumentationsregelwerk](#cloudogu-dokumentationsregelwerk) entspricht.
+  - Bei Abweichungen die Struktur entsprechend aufarbeiten. Dabei bestehende Strukturen bestehen lassen, aus den dort abgelegten Dateien auf den neuen `docs`-Ordner verlinken.
+  - Die alten Strukturen werden nach einer angemessenen Zeit (1 Jahr) gelöscht.
+
+## Cloudogu Dokumentationsregelwerk
```

Comment:
Would it not be good to tell external partners how to spell our products? So maybe we should at least include spelling policies for Cloudogu EcoSystem, Dogu and SCM-Manager? Maybe include User Management and Warp Menu as well?